### PR TITLE
Make certain arguments optional to allow creation of replicas

### PR DESCRIPTION
**Pull request by mihaiplesa**
## what

- Make certain arguments optional

## why

- Fix creation of replicas

## references

- Closes https://github.com/cloudposse/terraform-aws-rds/issues/137

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -24,14 +24,14 @@ resource "aws_db_instance" "default" {
 
   identifier            = module.this.id
   name                  = var.database_name
-  username              = var.database_user
-  password              = var.database_password
+  username              = length(var.replicate_source_db) > 0 ? null : var.database_user
+  password              = length(var.replicate_source_db) > 0 ? null : var.database_password
```
Comment (nitrocode) [inline main.tf:28]: can we set the comparison so null is the second argument

```suggestion
  password              = length(var.replicate_source_db) == 0 ? var.database_password : null
```

File: variables.tf [hunk 2]
```diff
@@ -82,7 +83,7 @@ variable "iops" {
 
 variable "allocated_storage" {
   type        = number
-  description = "The allocated storage in GBs"
+  description = "(Required unless a `snapshot_identifier` or `replicate_source_db` is provided) The allocated storage in GBs"
```
Comment (nitrocode) [inline variables.tf:86]: please use this for the others too

```suggestion
  description = "The allocated storage in GBs. Required unless a `snapshot_identifier` or `replicate_source_db` is provided."
```

File: variables.tf [hunk 3]
```diff
@@ -30,18 +30,19 @@ variable "associate_security_group_ids" {
 
 variable "database_name" {
   type        = string
+  default     = null
```
Comment (nitrocode) [inline variables.tf:33]: shouldn't the name be required

File: variables.tf (hunk 2 above)
Comment (mihaiplesa) [reply variables.tf]: Done.

**Comment by elkh510** [comment]
hi @jamengual @Benbentwo 
could you pls merge this one 
thank you!

File: main.tf [hunk 4]
```diff
@@ -24,14 +24,14 @@ resource "aws_db_instance" "default" {
 
   identifier            = module.this.id
   name                  = var.database_name
-  username              = var.database_user
-  password              = var.database_password
+  username              = length(var.replicate_source_db) > 0 ? null : var.database_user
```
Comment (nitrocode) [inline main.tf:27]: Lets set the logic so the falsey portion returns null. Please do this for all of the new logic to keep it consistent

```suggestion
  username              = length(var.replicate_source_db) == 0 ? var.database_user : null
```

File: variables.tf [hunk 5]
```diff
@@ -82,7 +83,7 @@ variable "iops" {
 
 variable "allocated_storage" {
   type        = number
-  description = "The allocated storage in GBs"
+  description = "The allocated storage in GBs. Required unless a `snapshot_identifier` or `replicate_source_db` is provided"
```
Comment (nitrocode) [inline variables.tf:86]:
```suggestion
  description = "The allocated storage in GBs. Required unless a `snapshot_identifier` or `replicate_source_db` is provided."
```

File: variables.tf [hunk 6]
```diff
@@ -30,19 +30,20 @@ variable "associate_security_group_ids" {
...
+  default     = null
   description = "The name of the database to create when the DB instance is created"
 }
 
 variable "database_user" {
   type        = string
-  default     = ""
-  description = "(Required unless a `snapshot_identifier` or `replicate_source_db` is provided) Username for the master DB user"
+  default     = null
+  description = "Username for the master DB user. Required unless a `snapshot_identifier` or `replicate_source_db` is provided"
```
Comment (nitrocode) [inline variables.tf:40]:
```suggestion
  description = "Username for the primary DB user. Required unless a `snapshot_identifier` or `replicate_source_db` is provided."
```

File: variables.tf [hunk 7]
```diff
@@ -30,19 +30,20 @@ variable "associate_security_group_ids" {
...
 }
 
 variable "database_password" {
   type        = string
-  default     = ""
-  description = "(Required unless a snapshot_identifier or replicate_source_db is provided) Password for the master DB user"
+  default     = null
+  description = "Password for the master DB user. Required unless a snapshot_identifier or replicate_source_db is provided"
```
Comment (nitrocode) [inline variables.tf:46]:
```suggestion
  description = "Password for the primary DB user. Required unless a `snapshot_identifier` or `replicate_source_db` is provided."
```

File: main.tf (hunk 4 above)
Comment (nitrocode) [reply main.tf]: @mihaiplesa please do this for all of the respective changes, not just this one. 😄

Comment (mihaiplesa) [reply main.tf]: thanks, let me know how it looks now :D

**Review by nitrocode — APPROVED** [review]

**Pull request merged**